test/morpho-blue/TestBlueSnippets.sol

```diff
@@ -112,72 +107,82 @@ contract TestIntegrationSnippets is BaseTest {
         assertEq(expectedTotalBorrow, morpho.totalBorrowAssets(id));
     }
 
-    function testBorrowAPY(Market memory market) public {
-        vm.assume(market.totalBorrowAssets > 0);
-        vm.assume(market.totalSupplyAssets >= market.totalBorrowAssets);
+    function testBorrowAPY(uint256 amountSupplied, uint256 amountBorrowed, uint256 timeElapsed, uint256 fee) public {
+        fee = bound(fee, 1, MAX_FEE);
 
-        uint256 borrowTrue = irm.borrowRate(marketParams, market).wTaylorCompounded(1);
-        uint256 borrowToTest = snippets.borrowAPY(marketParams, market);
+        _generatePendingInterest(amountSupplied, amountBorrowed, timeElapsed, fee);
+        morpho.accrueInterest(marketParams);
+
+        Market memory market = morpho.market(id);
+        uint256 expectedBorrowApy = irm.borrowRate(marketParams, market).wTaylorCompounded(365 days);
+        uint256 borrowApy = snippets.borrowAPY(marketParams, market);
 
-        assertEq(borrowTrue, borrowToTest, "Diff in snippets vs integration borrowAPY test");
+        assertEq(expectedBorrowApy, borrowApy, "Diff in snippets vs integration borrowAPY test");
+
+        if (expectedBorrowApy > 0) {
+            assertGt(borrowApy, 0, "The borrowAPY should be greater than zero but was found to be zero.");
+        }
     }
 
+    // Cover the idle market case - borrow
     function testBorrowAPYIdleMarket(Market memory market) public {
         MarketParams memory idleMarket;
         idleMarket.loanToken = address(loanToken);
 
-        uint256 borrowRate = snippets.borrowAPY(idleMarket, market);
+        uint256 borrowApy = snippets.borrowAPY(idleMarket, market);
 
-        assertEq(borrowRate, 0, "borrow rate");
+        assertEq(borrowApy, 0, "borrow rate");
     }
 
-    function testSupplyAPYEqual0(Market memory market) public {
-        vm.assume(market.totalBorrowAssets == 0);
-        vm.assume(market.totalSupplyAssets > 100000);
-        vm.assume(market.lastUpdate > 0);
-        vm.assume(market.fee < 1 ether);
-        vm.assume(market.totalSupplyAssets >= market.totalBorrowAssets);
+    // Cover the idle market case - supply
+    function testSupplyAPYIdleMarket(Market memory market) public {
```

Comment:
Same for this test, I can't see why we want to test that idle markets APY is zero